src/gen.cpp

```diff
@@ -603,8 +616,21 @@ int generate_cmake(const char *path, bool root) {
                                  [&](const std::string &, const std::vector<std::string> &includes) { inject_includes(includes); });
             gen.handle_condition(target.cmake_before, [&](const std::string &, const std::string &cmake) { inject_cmake(cmake); });
 
+            auto headers_var = target.name + "_HEADERS";
```

Comment:
I think the `headers` should just be appended to the `target_SOURCES` variable to actually also add them to `target_sources`. CMake doesn't have a concept of headers and when they are part of the `target_sources` you can get them from cmake etc.

Reply:
I see. The headers are now added to the sources.